tests/results/SourceRangeTest.py

```diff
@@ -81,6 +81,27 @@ def test_invalid_comparison(self):
         with self.assertRaises(TypeError):
             SourceRange(self.result_fileB_noline, self.result_fileB_line2) < 1
 
+    def test_affected_source(self):
+        file_dict = {abspath('file_name'): ('def fun():\n', '    x = 2  \n')}
+        sr1 = SourceRange.from_values('file_name', start_line=2, end_line=2)
+
+        self.assertEqual(sr1.affected_source(file_dict),
+                         ('    x = 2  \n',))
```

Comment:
don't repeat your doctests, that's just unneeded maintenance and code overhead. Remove all tests that are covered by your doctests.